khelo/lib/ui/flow/team/scanner/scanner_screen.dart

```diff
@@ -36,16 +42,29 @@ class _ScannerScreenState extends ConsumerState<ScannerScreen> {
   late final ScannerStateNotifier _notifier;
 
   void _observeBarcode() {
-    ref.listen(scannerStateNotifierProvider.select((value) => value.userId),
-        (prev, userId) async {
-      if (widget.addedMembers.contains(userId)) {
-        showSnackBar(context, context.l10n.add_team_member_already_added);
-      } else if (userId.isNotEmpty) {
-        final user =
-            await AppRoute.userDetail(userId: userId, showAddButton: true)
-                .push<UserModel?>(context);
-        if (mounted) {
-          context.pop(user);
+    ref.listen(scannerStateNotifierProvider.select((value) => value.scannedId),
+        (prev, scannedId) async {
+      if (widget.addedIds.contains(scannedId)) {
+        showSnackBar(
+            context,
+            widget.isForTeam
+                ? context.l10n.add_team_already_added
+                : context.l10n.add_team_member_already_added);
+      } else if (scannedId.isNotEmpty) {
+        if (widget.isForTeam) {
```

Comment:
instead of bool, we might want to use enum so that we can extend this implementation for tournament easily in future.